### PR TITLE
Maya - Easy turntable

### PR DESCRIPTION
**Pull request by antirotor**
## Feature

Easy turnable creator for Maya

🛠️ WIP

### CODE REVIEW

**Comment by mkolar** [comment]
Task linked: [OP-1431 Easy Turntable](https://app.clickup.com/t/6658547/OP-1431)

File: openpype/hosts/maya/plugins/create/create_turntable.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            # reference template file
+            cmds.file(path,
+                    reference=True,
+                    groupReference=True,
+                    namespace=namespace,
+                    groupName="_GRP",
+            )
+
+            # # get animation time and set key frames
+            cmds.setKeyframe(instance, t=1001, v=0.0, at="ry", itt="linear", ott="linear")
```
Comment (antirotor) [inline openpype/hosts/maya/plugins/create/create_turntable.py:47]: This won't work as start frame can be different then 1001 - it should be pulled from current maya timeline (that should be set to asset data)

File: openpype/hosts/maya/plugins/create/create_turntable.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                        asset=asset,
+                        options= {"useSelection":True},
+                        data={
+                            "frameStartHandle": 1001,
+                            "frameEndHandle": int(frames)+1001,
+                        }
+                        )
+
+            cmds.setAttr("defaultRenderGlobals.startFrame", 1001)
+            cmds.setAttr("defaultRenderGlobals.endFrame", int(frames)+1001)
```
Comment (BigRoy) [inline openpype/hosts/maya/plugins/create/create_turntable.py:76]: It feels weird to me to have this kind of logic in the Creator.

- What if after creation you want to update the start or end frame?
- What if after creation the model you're turntabling changed in size?
- Why does this need to mess with my render globals? Don't touch my render settings :)

I'd say, move this logic elsewhere, e.g. an Extractor or alike, where it temporarily creates this during the context. Or am I overthinking this?

**Comment by antirotor** [comment]
A agree. I would:

- change how this handles frame range
- change where it does it - but be aware that as it is using render family and render family by default render from published workfile, you would need to have those changes saved in  that workfile, so they can be temporary.
-  `turntable.mb` is pulling with itself dependencies that will stop Deadline for rendering until they are satisfied
-  During creation it is failing for me with:

image

**Comment by mkolar** [comment]
This doesn't have high priority on the internal backlog now, and we feel it will need to be reworked from the ground up. Closing it down as it's not really hitting the mark at the moment.

**Pull request closed without merging**